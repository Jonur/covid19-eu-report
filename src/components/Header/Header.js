import React from 'react';
import s from './Header.module.scss';

const Header = () => (
  <header className={s.appHeader}>
    <img src="european-union.png" alt="EU flag" title="EU flag" />
    <h1 className={s.appTitle}>EU COVID-19 Country Statistics</h1>
  </header>
);

export default Header;
