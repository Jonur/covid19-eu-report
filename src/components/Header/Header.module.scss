@import '../../media/colours';

.appHeader {
  display: flex;
  height: 80px;
  color: $app-white;
  padding: 1rem;
  border-radius: 2.56rem;
  background-color: $light-blue;
  margin: 0.5rem 0.5rem 0;

  .euStars {
    height: 100%;
    width: auto;
  }

  .headerText {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
    padding: 0 0 0 1rem;

    .appTitle {
      font-size: 1rem;
      margin: 0 0 0.25rem;
      font-weight: 400;
    }

    .lastUpdate {
      margin: 0;
      font-size: 0.875rem;
    }
  }

  @media (min-width: 768px) {
    width: 750px;
    margin: 0.5rem auto 0;
  }
}
