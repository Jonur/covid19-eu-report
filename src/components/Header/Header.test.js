import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';

describe('Header', () => {
  const props = { lastUpdate: '23 May 2020' };

  test('renders the Header component correctly', () => {
    const { container } = render(<Header {...props} />);
    expect(container).toMatchSnapshot();
  });
});
