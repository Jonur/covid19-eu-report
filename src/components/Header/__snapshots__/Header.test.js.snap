// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the Header component correctly 1`] = `
<div>
  <header
    class="appHeader"
  >
    <img
      alt="EU flag"
      src="european-union.png"
      title="EU flag"
    />
    <h1
      class="appTitle"
    >
      EU COVID-19 Country Statistics
    </h1>
    <p
      class="lastUpdate"
    >
      Data collected on 23 May 2020
    </p>
  </header>
</div>
`;
